src/sentry/search/eap/resolver.py

```diff
@@ -730,43 +730,44 @@ def resolve_function(self, column: str, match: Match | None = None) -> tuple[
                 f"Invalid number of arguments for {function}, was expecting {len(function_definition.required_arguments)} arguments"
             )
 
-        for index, argument in enumerate(function_definition.arguments):
-            if argument.ignored:
+        for index, argument_definition in enumerate(function_definition.arguments):
+            if argument_definition.ignored:
                 continue
-            if argument.validator is not None:
-                if not argument.validator(arguments[index]):
-                    raise InvalidSearchQuery(
-                        f"{arguments[index]} is not a valid argument for {function}"
-                    )
 
             if index < len(arguments):
-                if argument.is_attribute:
-                    parsed_argument, _ = self.resolve_attribute(arguments[index])
+                argument = arguments[index]
+                if argument_definition.validator is not None:
+                    if not argument_definition.validator(argument):
+                        raise InvalidSearchQuery(
+                            f"{argument} is not a valid argument for {function}"
+                        )
```

Comment:
Fyi, I moved the arguments validator under `if index < len(arguments)`. Otherwise we could run into `index out of range` issues. 

i.e we only need to validate arguments, if there's an argument that's being passed in